greenriver/hmis-warehouse
# Add `managed_in_version_control` flag to forms, and support duplicating them

## _Merging this PR_
- use the squash-merge strategy

## Description
Issue: https://github.com/open-path/Green-River/issues/6763
Frontend PR: https://github.com/greenriver/hmis-frontend/pull/973

The background here is that currently, the form builder lets you make new draft of system forms (say, the default Client form), and that causes problems when we try to deploy because the seed script doesn't expect it. The approach we went with was to disallow editing such forms in the config tool. (For now)

* Add `managed_in_version_control` flag to forms that are updated on deploy from the JsonForms seed script. This flag tells us that the form should NOT be edited in the Form Builder. (Very open to naming suggestions!! 👋  I felt that "system" was too generic and sometimes used to refer to forms that are required) 
* Add a mutation `CreateDuplicateFormDefinition` that allows admins to duplicate existing forms. This provides a path forward for **super admins** to create duplicates of forms that are "managed in version control" -- particularly helpful for QA.

## Type of change
New feature / bug fix

## Testing Instructions
* Run `rails driver:hmis:seed_definitions` to set the flag (tested with different clients)
* Pull down frontend branch https://github.com/greenriver/hmis-frontend/pull/973 with this, you should see:
  * system-managed forms are flagged and sorted to the end
  * when you open a system-managed form, there is an alert saying contact support to edit, and there is a Duplicate Form button
  * the Duplicate button works and creates a new form
  * if there were custom fields in the base form, the new form collects onto a new cded (not the same one) after publish. I tested this by temporarily turning on the "duplicate" button for a non-system form that I knew had CDEs on it.

## Checklist before requesting review
[//]: # (Remove any items that are not applicable)
